WooCommerce/StoreWidgets/StoreInfoProvider.swift

```diff
@@ -120,15 +120,31 @@ private extension StoreInfoProvider {
 
     /// Fetches the required dependencies from the keychain and the shared users default.
     ///
-    static func fetchDependencies() -> Dependencies? {
+    static func fetchDependencies(for selectedStore: IntentStore?) -> Dependencies? {
         let keychain = Keychain(service: WooConstants.keychainServiceName)
+        let storeID: Int64?
+        let storeName: String?
+
+        if let selectedStore {
+            if let storeIDString = selectedStore.identifier {
+                storeID = Int64(storeIDString)
+            } else {
+                storeID = nil
+            }
+            storeName = selectedStore.displayString
+        } else {
+            storeID = UserDefaults.group?[.defaultStoreID] as? Int64
+            storeName = UserDefaults.group?[.defaultStoreName] as? String
```

Comment:
Is this case still possible?

Reply:
`fetchDependencies(for: nil)` is used for placeholder state to show default store name (in the "add widget" flow).